internal/controller/controllers/debug_cmd.go

```diff
@@ -0,0 +1,116 @@
+package controllers
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/go-errors/errors"
+	"github.com/go-openapi/swag"
+	"github.com/openshift/oc/pkg/cli/debug"
+	"github.com/sirupsen/logrus"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+type DebugCommands interface {
+	RebootsForNode(nodeName string) (int, error)
+}
+
+type debugCommands struct {
+	log        logrus.FieldLogger
+	kubeconfig []byte
+}
+
+func NewDebugCommands(log logrus.FieldLogger, kubeconfig []byte) DebugCommands {
+	return &debugCommands{
+		log:        log,
+		kubeconfig: kubeconfig,
+	}
+}
+
+func defaultConfigFlags() *genericclioptions.ConfigFlags {
+	return genericclioptions.NewConfigFlags(true).
+		WithDeprecatedPasswordFlag().
+		WithDiscoveryBurst(350).
+		WithDiscoveryQPS(50.0)
+}
+
+func debugIostreams(command string, out, errOut io.Writer) genericclioptions.IOStreams {
+	return genericclioptions.IOStreams{
+		In:     bytes.NewReader([]byte(command)),
+		Out:    out,
+		ErrOut: errOut,
+	}
+}
+
+func debugFactory(kubeconfigFile string) kcmdutil.Factory {
+	flags := defaultConfigFlags()
+	flags.KubeConfig = swag.String(kubeconfigFile)
+	return kcmdutil.NewFactory(kcmdutil.NewMatchVersionFlags(flags))
+}
+
+func fatalHandler(log logrus.FieldLogger) func(string, int) {
+	return func(msg string, code int) {
+		err := errors.Errorf("fatal handler called: code %d, message %s", code, msg)
+		log.WithError(err).Error("fatalHandler")
+		panic(err)
+	}
+}
+
+func (d *debugCommands) kubeconfigFile() (string, error) {
+	var n int
+
+	f, err := os.CreateTemp("", "kubeconfig")
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+	numWritten := 0
+	for n, err = f.Write(d.kubeconfig[numWritten:]); err == io.ErrShortWrite; n, err = f.Write(d.kubeconfig[numWritten:]) {
+		numWritten += n
+	}
+	if err != nil {
+		return "", err
+	}
+	return f.Name(), nil
+}
+
+func (d *debugCommands) runDebug(command string, args ...string) (out string, errOut string, err error) {
+	kcmdutil.BehaviorOnFatal(fatalHandler(d.log))
```

Comment:
can you explain? cause it is running in routine?

Reply:
The default fatal behavior is to exit (which we don't want).  So I replaced with panic (and recover).
The expectation is that no code will be executed after fatal handler is called.

Reply:
Changed